velox/exec/Exchange.cpp

```diff
@@ -14,17 +14,29 @@
  * limitations under the License.
  */
 #include "velox/exec/Exchange.h"
+#include <algorithm>
+#include <random>
 #include "velox/exec/Task.h"
 
 namespace facebook::velox::exec {
 
+void Exchange::addTaskIds(std::vector<std::string> taskIds) {
+  auto rng = std::default_random_engine{};
+  std::shuffle(std::begin(taskIds), std::end(taskIds), rng);
+  for (std::string taskId : taskIds) {
+    exchangeClient_->addRemoteTaskId(taskId);
+    ++stats_.wlock()->numSplits;
+  }
+}
+
 bool Exchange::getSplits(ContinueFuture* future) {
   if (!processSplits_) {
     return false;
   }
   if (noMoreSplits_) {
     return false;
   }
+  std::vector<std::string> taskIds;
```

Comment:
Does Presto Java have the same algorithm?

Reply:
We try this optimization for native. It is always help to avoid reading from the same source in a distributed system.